dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -1537,6 +1540,93 @@ def test_profiler_factory_class_creates_correct_profiler(self, *mocks):
         self.assertIsInstance(Profiler(dict_df), UnstructuredProfiler)
         self.assertIsInstance(Profiler(dict_ser), UnstructuredProfiler)
 
+    def test_save_and_load_structured(self):
+        datapth = "dataprofiler/tests/data/"
+        test_files = ["csv/guns.csv", "csv/iris.csv"]
+
+        def _clean_report(report):
+            data_stats = report["data_stats"]
+            for key in data_stats:
+                stats = data_stats[key]["statistics"]
+                if "histogram" in stats:
+                    if "bin_counts" in stats["histogram"]:
+                        stats["histogram"]["bin_counts"] = \
+                            stats["histogram"]["bin_counts"].tolist()
+                    if "bin_edges" in stats["histogram"]:
+                        stats["histogram"]["bin_edges"] = \
+                            stats["histogram"]["bin_edges"].tolist()
+            return report
+
+        for test_file in test_files:
+            # Create Data and StructuredProfiler objects
+            data = dp.Data(os.path.join(datapth, test_file))
+            save_profile = dp.StructuredProfiler(data)
+
+            # Save and Load profile with Mock IO
+            with mock.patch('builtins.open') as m:
+                mock_file = setup_save_mock_open(m)
+                save_profile.save()
+                mock_file.seek(0)
+                with mock.patch('dataprofiler.profilers.profile_builder.'
+                                'DataLabeler'):
+                    load_profile = dp.Profiler.load("mock.pkl")
+
+            # Check that reports are equivalent
+            save_report = _clean_report(save_profile.report())
+            load_report = _clean_report(load_profile.report())
+            self.assertDictEqual(save_report, load_report)
+
+    def test_save_and_load_unstructured(self):
+        data_folder = "dataprofiler/tests/data/"
+        test_files = ["txt/code.txt", "txt/sentence-10x.txt"]
+
+        for test_file in test_files:
+            # Create Data and StructuredProfiler objects
+            data = dp.Data(os.path.join(data_folder, test_file))
+            save_profile = UnstructuredProfiler(data)
+
+            # address case where empty_line_count is 0
+            save_profile._empty_line_count = 1
+
+            # store the expected data_labeler
+            data_labeler = save_profile.options.data_labeler.data_labeler_object
```

Comment:
Why don't you do these data_labeler checks in the structured test function?

Reply:
these are copies of other tests where all i did was swap out the single line
`load_profile = Profiler.load("mock.pkl")`

Written by two diff ppl and one was written without the extra case